### PR TITLE
feat(providers/api/marketmap): Add raydium to dydx market-map provider [BLO-1084]

### PR DESCRIPTION
**Pull request by g3ndo9kar9**

### CODE REVIEW

File: providers/apis/dydx/parse.go [hunk 1]
```diff
@@ -180,35 +185,118 @@ func (h *APIHandler) ConvertExchangeConfigJSON(
...
+			return "", fmt.Errorf("failed to marshal %s provider metadata for ticker %s: %w", raydium.Name, ticker, err)
+		}
+
+		return string(bz), nil
+	}
+
+	return "", nil
+}
+
+// raydiumTickerMetadata extracts a raydiumTickerMetadata from a dydx ticker.
```
Comment (davidterpay) [inline providers/apis/dydx/parse.go]: can we move this to the raydium provider and have it be a helper function.

Comment (davidterpay) [reply providers/apis/dydx/parse.go]: Can we also document what the expected ticker string looks like?

File: providers/apis/dydx/parse.go [hunk 2]
```diff
@@ -180,35 +185,118 @@ func (h *APIHandler) ConvertExchangeConfigJSON(
...
+func raydiumTickerMetadata(ticker string) (raydium.TickerMetadata, error) {
+	// split fields by separator and expect there to be at least 6 values
+	fields := strings.Split(ticker, "/")
```
Comment (davidterpay) [inline providers/apis/dydx/parse.go]: dYdX uses "-" as a delimeter btw.

Comment (davidterpay) [reply providers/apis/dydx/parse.go]: We should assume that

File: providers/apis/dydx/parse.go [hunk 3]
```diff
@@ -180,35 +185,118 @@ func (h *APIHandler) ConvertExchangeConfigJSON(
...
 			Invert:          cfg.Invert,
 			NormalizeByPair: normalizeByPair,
+			Metadata_JSON:   metaData,
 		})
 	}
 
 	return providers, nil
 }
 
+// ExtractMetadataFromTicker extracts json-metadata from a ticker, based on the exchange. Specifically,
```
Comment (davidterpay) [inline providers/apis/dydx/parse.go]: This is likely not going to be raydium specific, could we move these docs to raydium

**Review by aljo242 — APPROVED** [review]
Yay

**Review by Eric-Warehime — APPROVED** [review]

**Pull request merged**